jina/peapods/runtimes/container/__init__.py

```diff
@@ -163,6 +164,36 @@ def _docker_run(self, replay: bool = False):
                     'mode': 'rw',
                 }
 
+        device_requests = []
+        if self.args.gpus:
+            _gpus = {
+                'count': 0,
+                'capabilities': ['gpu'],
+                'device': [],
+                'driver': '',
+            }
+            gpu_args = self.args.gpus[0]
+            for gpu_arg in gpu_args.split(','):
+                if gpu_arg == 'all':
+                    _gpus['count'] = -1
+                if gpu_arg.isdigit():
+                    _gpus['count'] = int(gpu_arg)
+                if '=' in gpu_arg:
+                    gpu_arg_key, gpu_arg_value = gpu_arg.split('=')
+                    if gpu_arg_key in _gpus.keys():
+                        if isinstance(_gpus[gpu_arg_key], list):
+                            _gpus[gpu_arg_key].append(gpu_arg_value)
+                        else:
+                            _gpus[gpu_arg_key] = gpu_arg_value
+            device_requests = [
+                docker.types.DeviceRequest(
+                    count=_gpus['count'],
+                    driver=_gpus['driver'],
+                    device_ids=_gpus['device'],
+                    capabilities=[_gpus['capabilities']],
+                )
+            ]
+
```

Comment:
remove gpus from args before doing the container run for better backwards compatibility with jinahub on older jina versions